### ISSUE TITLE
FR: Multiple search bases?

### ISSUE THREAD

**Issue by nathanperkins** [etianen/django-python3-ldap#82]
I'm not sure if this is possible but I want to pull from two OUs that are in different paths.

Is there a way to accomplish this? Otherwise, maybe it would be a decent feature to add?

**Comment by etianen** [comment]
Can you simply specify a higher OU? django-python3-ldap searches recursively.

**Comment by nathanperkins** [comment]
For my use case, its not easy. We have non-perm users in a different OU from our perm users but the parent OU also has administrative and test users that I don't want to sync into my app. Other apps depend on our OU structure so adjusting the OUs could break other apps.

**Comment by etianen** [comment]
I think that your solution is to specify a common ancestor of the search
tree, then add some search filters, like this:

https://github.com/etianen/django-python3-ldap#custom-user-filters

**Comment by sherdwhite** [comment]
How exactly does the custom user filter work?  I am a bit confused regarding the "path.to.your.custom_format_search_filters".  Can you have the custom_format_search_filters in the settings.py file?  The comment "# pay/to/your.py" means?  I am able to successfully pull down all users of a certain group but the way our AD is set up, I need to also pull from different OU and wanted to filter using a department field so that I only grab users from a particular department.   So my function would look like:  

from django_python3_ldap.utils import format_search_filters

def custom_format_search_filters(ldap_fields):
    # Add in simple filters.
    ldap_fields["department"] = "Engineering"
    # Call the base format callable.
    search_filters = format_search_filters(ldap_fields)
    # Advanced: apply custom LDAP filter logic.
    # search_filters.append("(|(memberOf=groupA)(memberOf=GroupB))")
    # All done!
    return search_filters

**Comment by sherdwhite** [comment]
I am looking at this incorrectly.  I LDAP is just authenticating the user.  Is it possible to get access to a custom attribute from AD so that I can restrict login based on this AD attribute?

**Comment by etianen** [comment]
I think you do want to use custom search filters here.

There was a typo in the instructions, which may have confused you. I've
updated the instructions here:

https://github.com/etianen/django-python3-ldap#custom-user-filters

**Comment by sherdwhite** [comment]
Thanks, I think what I was trying to do worked better through the ldap3 library.  I am using django-python3-ldap for AD authentication for all my web apps, but each individual app is using ldap3 to verify credentials for each user so that I have a bit more control in each app to tailor access.

**Comment by komaxko** [comment]
Greetings! 
Have same problem, but i need specify a couple of "memberof" params in.

so, my module looks like:
from django_python3_ldap.utils import format_search_filters


def custom_format_search_filters(ldap_fields):
    # Add in simple filters.
    # ldap_fields["memberOf"] = ""
    #ldap_fields["memberof"] = "ou=users, ou=it"
    # ldap_fields["ipPhone"] = "0000"
    # Call the base format callable.
    search_filters = format_search_filters(ldap_fields)
    # Advanced: apply custom LDAP filter logic.
    # search_filters.append("(|(ou=users, ou=it))")

    # (&(objectCategory=user)(|(ou=staff)(ou=vendors)))
    # All done!
    return search_filters

But if i uncomment any string - I can't login. Just ipPhone  works, cause it has been sepcified in AD attrs.

And i can't find attr "memberOf" in all attrs for a user (object) in AD.

**Comment by klainn** [comment]
I know this is an old issue, but i'm a new user of this plugin and am having the same issue.

I've successfully got a group setup and it does sync users from that group properly, as long as they are in the same domain.

We're setup as such:
My search filter works if I set it to "DC=usa,DC=company,DC=com"
and I get everyone in my custom filter group who is in DC=usa,DC=company,DC=com. However I have people in different domains in the same group. For example DC=india,DC=company,DC=com. With my search filter in settings.py as DC=usa,DC=company,DC=com, the DC=india folks are ignored.

If I change my search filter in settings.py to DC=company,DC=com, which would be the obvious thing to do, I get no error, but the additional users do not get sync'd.
